### PR TITLE
Documentation and re-arrange members

### PR DESCRIPTION
**Pull request by ciuckc**
- documentation for config
- more documentation for config.
- Change: isDirective symbol to isServerDirective.
- Config class description
- Re-arrange members of ConfigServer
- brace init server name to localhost

### CODE REVIEW

File: srcs/config/ConfigParse.cc [hunk 1]
```diff
@@ -42,7 +42,7 @@ ConfigParse::ConfigParse(const Lines& file_data) : lines_(file_data), map_(), lo
 
   this->loc_map_ = {{"index", &ConfigParse::indexParse},       {"autoindex", &ConfigParse::autoIndexParse},
                     {"root", &ConfigParse::rootParse},         {"allowed_methods", &ConfigParse::allowedMethodsParse},
-                    {"redirect", &ConfigParse::redirectParse}, {"upload_dir", &ConfigParse::uploadDirParse}};
+                    {"redirect", &ConfigParse::redirectParse}, {"cgi_upload_dir", &ConfigParse::uploadCgiDirParse}};
```
Comment (Lucwousin) [inline srcs/config/ConfigParse.cc]: it feels kind of weird to me to call this directory cgi_upload_dir when cgi does not handle the uploads

File: srcs/config/ConfigServer.cc [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
 
 ConfigServer::ConfigServer()
-    : port_(),
-      client_max_body_size_(std::numeric_limits<size_t>::max()),
-      routes_(),
+    : routes_(),
       error_pages_(),
-      files_() {}
+      files_{},
+      server_name_{"localhost"},
```
Comment (Lucwousin) [inline srcs/config/ConfigServer.cc]: We can't initialize server_name_ to localhost, an empty server_name_ is allowed and we'll run into problems with servers using the same host:port combo (check out ListenSocket::addConfig)

File: srcs/config/ConfigServer.h [hunk 3]
```diff
@@ -35,10 +39,11 @@ class ConfigServer {
...
-  std::vector<std::string> server_name_;
-  uint16_t port_;
-  std::size_t client_max_body_size_;
-  routes_t routes_;
-  std::map<int, std::string> error_pages_;
-  index_files_t files_;
+  routes_t routes_{};
+  std::map<int, std::string> error_pages_{};
+  index_files_t files_{};
+  std::vector<std::string> server_name_{"localhost"};
```
Comment (Lucwousin) [inline srcs/config/ConfigServer.h]: You're initializing everything twice now by defining the constructor and the initial values

**Review by Lucwousin — CHANGES_REQUESTED** [review]

File: srcs/config/ConfigParse.cc (hunk 1 above)
Comment (ciuckc) [reply srcs/config/ConfigParse.cc]: Oh this was a remnant from before

**Review by ciuckc — COMMENTED** [review]
Fix!

**Comment by Lucwousin** [comment]
now it's missing the entire upload_dir no? :joy:

**Pull request closed without merging**